unittests/utils/tensor_tester.cpp

```diff
@@ -0,0 +1,282 @@
+  data[1] = {0.0, 1.0};
+  data[2] = {0.0, -1.0};
+  data[3] = {1.0, 0.0};
+
+  cudaq::tensor t(data, shape);
+
+  EXPECT_EQ(t.rank(), 2);
+  EXPECT_EQ(t.size(), 4);
+  EXPECT_EQ(t.shape(), shape);
+
+  // Check if data is correctly stored
+  EXPECT_EQ(t.at({0, 0}), std::complex<double>(1.0, 0.0));
+  EXPECT_EQ(t.at({0, 1}), std::complex<double>(0.0, 1.0));
+  EXPECT_EQ(t.at({1, 0}), std::complex<double>(0.0, -1.0));
+  EXPECT_EQ(t.at({1, 1}), std::complex<double>(1.0, 0.0));
+}
+
+TEST(TensorTest, AccessElements) {
+  std::vector<std::size_t> shape = {2, 3};
+  cudaq::tensor t(shape);
+
+  // Set values
+  t.at({0, 0}) = {1.0, 0.0};
+  t.at({0, 1}) = {0.0, 1.0};
+  t.at({1, 2}) = {-1.0, 0.0};
+
+  // Check values
+  EXPECT_EQ(t.at({0, 0}), std::complex<double>(1.0, 0.0));
+  EXPECT_EQ(t.at({0, 1}), std::complex<double>(0.0, 1.0));
+  EXPECT_EQ(t.at({1, 2}), std::complex<double>(-1.0, 0.0));
+}
+
+TEST(TensorTest, CopyData) {
+  std::vector<std::size_t> shape = {2, 2};
+  std::vector<std::complex<double>> data = {
+      {1.0, 0.0}, {0.0, 1.0}, {0.0, -1.0}, {1.0, 0.0}};
+  cudaq::tensor t(shape);
+
+  t.copy(data.data(), shape);
+
+  EXPECT_EQ(t.at({0, 0}), std::complex<double>(1.0, 0.0));
+  EXPECT_EQ(t.at({0, 1}), std::complex<double>(0.0, 1.0));
+  EXPECT_EQ(t.at({1, 0}), std::complex<double>(0.0, -1.0));
+  EXPECT_EQ(t.at({1, 1}), std::complex<double>(1.0, 0.0));
+}
+
+TEST(TensorTest, TakeData) {
+  std::vector<std::size_t> shape = {2, 2};
+  auto data = std::make_unique<std::complex<double>[]>(4);
+  const std::vector<std::complex<double>> idata{
+      {1.0, 0.0}, {0.0, 1.0}, {0.0, -1.0}, {1.0, 0.0}};
+  std::copy(idata.begin(), idata.end(), data.get());
+  cudaq::tensor t(shape);
+
+  t.take(data, shape);
+
+  EXPECT_EQ(t.at({0, 0}), std::complex<double>(1.0, 0.0));
+  EXPECT_EQ(t.at({0, 1}), std::complex<double>(0.0, 1.0));
+  EXPECT_EQ(t.at({1, 0}), std::complex<double>(0.0, -1.0));
+  EXPECT_EQ(t.at({1, 1}), std::complex<double>(1.0, 0.0));
+
+  // Note: We don't delete data here as the tensor now owns it
+}
+
+TEST(TensorTest, BorrowData) {
+  std::vector<std::size_t> shape = {2, 2};
+  std::vector<std::complex<double>> data = {
+      {1.0, 0.0}, {0.0, 1.0}, {0.0, -1.0}, {1.0, 0.0}};
+  cudaq::tensor t(shape);
+
+  t.borrow(data.data(), shape);
+
+  EXPECT_EQ(t.at({0, 0}), std::complex<double>(1.0, 0.0));
+  EXPECT_EQ(t.at({0, 1}), std::complex<double>(0.0, 1.0));
+  EXPECT_EQ(t.at({1, 0}), std::complex<double>(0.0, -1.0));
+  EXPECT_EQ(t.at({1, 1}), std::complex<double>(1.0, 0.0));
+}
+
+TEST(TensorTest, InvalidAccess) {
+  std::vector<std::size_t> shape = {2, 2};
+  cudaq::tensor t(shape);
+
+  std::exception_ptr eptr;
+  try {
+    t.at({2, 0});
+  } catch (...) {
+    eptr = std::current_exception();
+    auto info = eptr.__cxa_exception_type();
+    std::cout << "Caught exception of type : " << info->name() << std::endl;
+    throw std::runtime_error(info->name());
+  }
+
+  EXPECT_THROW(t.at({2, 0}), std::runtime_error);
```

Comment:
This test fails in CI `Create CUDA Quantum installer (xxx) / Build CUDA Quantum assets` task

Reply:
Trying to disable.